File: c++/js/countSort.js
function countSort(a, k, n){
	var c=[];
	var b=[];
	for(var i = 0; i <= k; i++){
		c[i] = 0;                   //c数组的范围是a数组中最大数值
	}
	for(var j = 0; j < n; j ++){    //a数组默认是从0开始
		c[a[j]] ++;
	}
	for(var l = 1; l <= k; l++){
		c[l] = c[l]+c[l-1];    //统计小于等于该元素的个数 
	}
	for(var v = n-1; v >= 0; v--){
		b[c[a[v]]-1] = a[v];
		c[a[v]] --;
	}
	return b;
}

console.log(countSort([10,9,8,7,6,5,4,3,2,1], 10, 10));

File: c++/countSort.cpp
#include <istream>
#include <cstdio>
#include <algorithm>
using namespace std;
int c[1000]; //1000代表了a数组中元素的最大值 
int b[1000];
int a[1000];
void countSort(int *a, int *b, int k, int n)
{
	for(int i = 0; i <= k; i++)
	{
		c[i] = 0;
	}
	for(int j = 0; j < n; j++)
	{
		c[a[j]] ++;
	}
	for(int i = 1; i <= k; i++)
	{
		c[i] = c[i]+ c[i-1];
	}
	for(int j = n-1; j >= 0; j--)
	{
		b[c[a[j]]] = a[j];
		c[a[j]] --;
	}
}
int main()
{
	int n;
	int max_x = -100;
	scanf("%d", &n);
	for(int i = 0; i < n; i++)
	{
		scanf("%d", &a[i]);
		if(a[i]>max_x)
		max_x = a[i];
	}
	countSort(a, b, max_x, n);
	for(int i = 1; i <= n; i++)
	{
		printf("%d ", b[i]);
	}
	return 0;
}
